pkg/auth/auth.go

```diff
@@ -43,6 +44,41 @@ func CheckAuthFile(authfile string) error {
 	return nil
 }
 
+// Mirrors --authfile to a tmpfile if --authfile points to a
+// file descriptor instead of actual file on filesystem
+// reason is file descriptor could vanish mid-session if consumed
+// once, check https://github.com/containers/buildah/issues/3070
+func CreateTempAuthFileIfNeeded(authfile string) string {
+	if strings.HasPrefix(authfile, "/dev/fd") {
+		b, err := ioutil.ReadFile(authfile)
+		if err != nil {
+			// if anything goes wrong return authfile
+			// and let c/common/auth handle failues later
+			return authfile
+		}
+		tmpfile, err := ioutil.TempFile(os.TempDir(), "temp-common-auth")
+		if err != nil {
+			return authfile
+		}
+		if _, err := tmpfile.Write(b); err != nil {
+			// if anything goes wrong return authfile
+			// and let c/common/auth handle failues later
+			return authfile
+		}
+
+		return tmpfile.Name()
+	}
+	return authfile
+}
+
+// Removes any temp auth file if it was created in the first place
+func CleanTempAuthFiles() {
+	files, _ := filepath.Glob(filepath.Join(os.TempDir(), "temp-common-auth"))
+	for _, f := range files {
+		os.Remove(f)
+	}
+}
```

Comment:
Sure can fix this @Luap99  regarding your query about having this in `c/image` one problem with handling it with c/image instead of buildah or podman is that we will have to handle the states (cleanups,creation) of `temp auth file` from c/image however c/image is not aware about when a `buildah` session is started and when is it going to end. So there can be cases where `c/image` actually creates the file and deletes while `buildah` or `podman` would still need it for mult-stage builds.

So I think state of temp-auth-file is better managed by `podman` or `buildah`.

Reply:
This function is a blocker to me.  We cannot delete "random" files. Each caller should take care of doing that.